File: fitrbackend/src/main/java/com/example/fitrbackend/dto/PlanExerciseResponse.java
package com.example.fitrbackend.dto;

import lombok.Getter;

public class PlanExerciseResponse {

    @Getter
    private final long id;
    @Getter
    private final long plan_day_id;
    @Getter
    private final long exercise_id;
    @Getter
    private final int targetSets;
    @Getter
    private final int targetReps;
    @Getter
    private final int targetDurationSeconds;
    @Getter
    private final float targetDistance;
    @Getter
    private final float targetCalories;

    public PlanExerciseResponse(long id, long plan_day_id, long exercise_id, int targetSets, int targetReps, int targetDurationSeconds, float targetDistance, float targetCalories) {
        this.id = id;
        this.plan_day_id = plan_day_id;
        this.exercise_id = exercise_id;
        this.targetSets = targetSets;
        this.targetReps = targetReps;
        this.targetDurationSeconds = targetDurationSeconds;
        this.targetDistance = targetDistance;
        this.targetCalories = targetCalories;
    }
}

File: fitrfrontend/fitrfrontend/API/PlanExerciseResponse.swift
//
//  PlanExerciseResponse.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

struct PlanExerciseResponse: Codable, Identifiable {
  let id: Int64
  let planDayId: Int64
  let exerciseId: Int64
  let targetSets: Int
  let targetReps: Int
  let targetDurationSeconds: Int
  let targetDistance: Float
  let targetCalories: Float
  let targetWeight: Float?

  enum CodingKeys: String, CodingKey {
    case id
    case planDayId = "plan_day_id"
    case exerciseId = "exercise_id"
    case targetSets
    case targetReps
    case targetDurationSeconds
    case targetDistance
    case targetCalories
    case targetWeight
  }
}
